src/routes/class-base-buckaroo-route.php

```diff
@@ -0,0 +1,154 @@
+  ) {
+    $this->config                          = $config;
+    $this->buckaroo                        = $buckaroo;
+    $this->buckaroo_response_handler_route = $buckaroo_response_handler_route;
+    $this->hmac                            = $hmac;
+    $this->basic_captcha                   = $basic_captcha;
+  }
+
+  /**
+   * We need to define redirect URLs so that Buckaroo redirects the user to our buckaroo-response-handler route
+   * which might run some custom logic and then redirect the user to the actual redirect URL as defined in the form's
+   * options.
+   *
+   * @param array $params Array of WP_REST_Request params.
+   * @return array
+   */
+  protected function set_redirect_urls( array $params ): array {
+
+    // Now let's define all Buckaroo-recognized statuses for which we need to provide redirect URLs.
+    $statuses = [
+      Buckaroo_Response_Handler_Route::STATUS_SUCCESS,
+      Buckaroo_Response_Handler_Route::STATUS_CANCELED,
+      Buckaroo_Response_Handler_Route::STATUS_ERROR,
+      Buckaroo_Response_Handler_Route::STATUS_REJECT,
+    ];
+
+    // Now let's build redirect URLs (to buckaroo-response-handler middleware route) for each status.
+    $redirect_urls = [];
+    $base_url      = \home_url( $this->buckaroo_response_handler_route->get_route_uri() );
+    foreach ( $statuses as $status_value ) {
+      $url_params = $params;
+      $url_params[ Buckaroo_Response_Handler_Route::STATUS_PARAM ] = $status_value;
+      $url = \add_query_arg( array_merge(
+        $url_params,
+        [ HMAC::AUTHORIZATION_KEY => rawurlencode( $this->hmac->generate_hash( $url_params, $this->generate_authorization_salt_for_response_handler() ) ) ]
+      ), $base_url );
+
+      $redirect_urls[] = $url;
+    }
+
+    $this->buckaroo->set_redirect_urls( ...$redirect_urls );
+
+    return $params;
+  }
+
+  /**
+   * Set Buckaroo to test mode if test param provided.
+   *
+   * @param array $params Request params.
+   * @return void
+   */
+  protected function set_test_if_needed( array $params ): void {
+    if ( isset( $params[ self::TEST_PARAM ] ) && filter_var( $params[ self::TEST_PARAM ], FILTER_VALIDATE_BOOL ) ) {
+      $this->buckaroo->set_test();
+    }
+  }
+
+  /**
+   * Define authorization salt used for request to response handler.
+   *
+   * @return string
+   */
+  protected function generate_authorization_salt_for_response_handler(): string {
+    return \apply_filters( Filters::BUCKAROO, 'secret_key' ) ?? 'invalid-salt-for-emandate-handler';
+  }
+
+  /**
+   * Method that returns rest response
+   *
+   * @param  \WP_REST_Request $request Data got from endpoint url.
+   *
+   * @return WP_REST_Response|mixed If response generated an error, WP_Error, if response
+   *                                is already an instance, WP_HTTP_Response, otherwise
+   *                                returns a new WP_REST_Response instance.
+   */
+  abstract public function route_callback( \WP_REST_Request $request );
```

Comment:
Shouldn't this be defined by the REST callable interface?

Reply:
Yes, this is the abstract route for routes that handle Buckaroo things which extends `Base_Route` (which implements `Callable_Route`). Because of this it needs to have a `route_callback` defined (but as `abstract` because every Buckaroo route that extends this one will have a different `route_callback` implementation).